script/tool/test/analyze_command_test.dart

```diff
@@ -519,42 +589,945 @@ packages/package_a/$file
 
         expect(
             output,
-            isNot(containsAllInOrder(<Matcher>[
+            containsAllInOrder(<Matcher>[
               contains('Running for package_a'),
-            ])));
+            ]));
+      });
+
+      const List<String> files = <String>[
+        'foo.java',
+        'foo.kt',
+        'foo.m',
+        'foo.swift',
+        'foo.c',
+        'foo.cc',
+        'foo.cpp',
+        'foo.h',
+      ];
+      for (final String file in files) {
+        test('skips command for changes to non-Dart source $file', () async {
+          createFakePackage('package_a', packagesDir);
+
+          gitProcessRunner.mockProcessesForExecutable['git-diff'] =
+              <FakeProcessInfo>[
+            FakeProcessInfo(MockProcess(stdout: '''
+packages/package_a/$file
+''')),
+          ];
+
+          final List<String> output =
+              await runCapturingPrint(runner, <String>['analyze']);
+
+          expect(
+              output,
+              isNot(containsAllInOrder(<Matcher>[
+                contains('Running for package_a'),
+              ])));
+          expect(
+              output,
+              containsAllInOrder(<Matcher>[
+                contains('SKIPPING ALL PACKAGES'),
+              ]));
+        });
+      }
+
+      test('skips commands if all files should be ignored', () async {
+        createFakePackage('package_a', packagesDir);
+
+        gitProcessRunner.mockProcessesForExecutable['git-diff'] =
+            <FakeProcessInfo>[
+          FakeProcessInfo(MockProcess(stdout: '''
+README.md
+CODEOWNERS
+packages/package_a/CHANGELOG.md
+''')),
+        ];
+
+        final List<String> output =
+            await runCapturingPrint(runner, <String>['analyze']);
+
+        expect(
+            output,
+            isNot(containsAllInOrder(<Matcher>[
+              contains('Running for package_a'),
+            ])));
         expect(
             output,
             containsAllInOrder(<Matcher>[
               contains('SKIPPING ALL PACKAGES'),
             ]));
       });
-    }
+    });
+  });
 
-    test('skips commands if all files should be ignored', () async {
-      createFakePackage('package_a', packagesDir);
+  group('gradle lint', () {
```

Comment:
This group is the old `lint-android` tests, with minor string changes where I changed some logging messages.